### PR TITLE
Number of Lines

### PR DESCRIPTION
**Pull request by Romucmad**
#4  
HighlightedFile can have numberOfLines. By default it's None what will show all lines otherwise just number in given argument. Negative number will   call sys.exit(1)

### CODE REVIEW

**Comment by IanMayo** [comment]
Nicely done. But there are two methods that are almost identical. You should refactor them - maybe so that one is a "thin wrapper" for the other.

File: data_highlight/highlighter.py [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
 from .support.color_picker import  hexColorFor, meanColorFor, colorFor
+
 class HighlightedFile():
     """
     class that can load/tokenize a datafile, record changes to the file,
     then export a highlighted version of the file that indicates extraction
     """
 
-    def __init__(self, filename: str):
+    def __init__(self, filename: str,numberOfLines=None):
```
Comment (IanMayo) [inline data_highlight/highlighter.py]: we need a comment for the new parameter, please.

File: data_highlight/highlighter.py [hunk 2]
```diff
@@ -105,3 +92,67 @@ def export(self, filename: str):
...
+        stringToChar = '\n'.join(str(e) for e in strLines)
+        self.chars = [None] * len(stringToChar)
+        # initialise the char index
+        charCtr = 0
+        for char in stringToChar:
+            # put letter into a struct
+            charInd = CharIndex(char)
+            self.chars[charCtr] = charInd
+            charCtr += 1
+        counter=0
```
Comment (IanMayo) [inline data_highlight/highlighter.py]: `counter` isn't used.

File: data_highlight/highlighter.py [hunk 3]
```diff
@@ -105,3 +92,67 @@ def export(self, filename: str):
...
+            charInd = CharIndex(char)
+            self.chars[charCtr] = charInd
+            charCtr += 1
+
+            # ok, break the file into self-aware lines
+        lineCtr = 0
+        lines = []
+        strLines = sampleLines.splitlines()
+
+        stringToChar = ''.join(str(e) for e in strLines)
```
Comment (IanMayo) [inline data_highlight/highlighter.py]: `stringToChar` isn't used.

File: data_highlight/highlighter.py [hunk 4]
```diff
@@ -105,3 +92,67 @@ def export(self, filename: str):
...
+        """
+
+        with open(self.filename, 'r') as file:
+            sampleLines = file.read()
+
+        lineCtr = 0
+        lines = []
+        strLines = sampleLines.splitlines()
+        strLines= strLines[0:self.numberOfLines]
+        stringToChar = '\n'.join(str(e) for e in strLines)
```
Comment (IanMayo) [inline data_highlight/highlighter.py]: would you mind adding a comment, just to explain what we're doing with `stringToChar` above.  Hold on - are we separating the lines with `newline` chars?

File: test/test_numbers.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+path = os.path.abspath(__file__)
+dir_path = os.path.dirname(path)
+TEST_FILE = os.path.join(dir_path, "reptest1.rep")
+ERROR_TEST_FILE = os.path.join(dir_path, "reptest2.rep")
+
+
+DATA_FILE = 'data_highlight/file.txt'
+
+class SimpleTest(unittest.TestCase):
```
Comment (IanMayo) [inline test/test_numbers.py]: Could you rename this to LineCountTest please?

File: test/test_numbers.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import os
```
Comment (IanMayo) [inline test/test_numbers.py]: I'm pretty sure we don't use `os`

I also don't think we use lines 5-8, where we get at `reptest` files.

File: test/test_numbers.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    def tearDown(self):
+        pass
+
+    ####################
+    #### file tests ####
+    ####################
+
+    def test_number_of_lines(self):
+
+        dataFile = HighlightedFile('data_highlight/file.txt',2)
```
Comment (IanMayo) [inline test/test_numbers.py]: we don't need to use this string filename when we have a constant for it at line 11.

**Comment by IanMayo** [comment]
@Romucmad - could you also do a test that demonstrates things don't break if we request more lines than there are in the data-file?

File: test/test_numbers.py (hunk 7 above)
Comment (Romucmad) [reply test/test_numbers.py]: Oh okay , I didn't mention contstant.

Comment (Romucmad) [reply test/test_numbers.py]: Okay Got it

**Comment by Romucmad** [comment]
Yes I will write the test.

**Pull request merged**